packages/typespec-client-generator-core/test/clients/structure.test.ts

```diff
@@ -580,14 +580,48 @@ it("namespace", async () => {
     @server("http://localhost:3000", "endpoint")
     @useAuth(ApiKeyAuth<ApiKeyLocation.header, "x-ms-api-key">)
     @service
-    namespace My.Service {};
+    namespace My.Service;
+    op func(): void;
   `);
   const sdkPackage = runnerWithCore.context.sdkPackage;
   strictEqual(sdkPackage.clients.length, 1);
   const clientOne = sdkPackage.clients.filter((c) => c.name === "ServiceClient")[0];
   strictEqual(clientOne.namespace, "My.Service");
 });
 
+it("model-only namespace should be filtered out", async () => {
+  const runnerWithCore = await createSdkTestRunner({
+    librariesToAdd: [AzureCoreTestLibrary],
+    autoUsings: ["Azure.Core"],
+    emitterName: "@azure-tools/typespec-java",
+  });
+  await runnerWithCore.compile(`
+    namespace Foo {
+      model B {}
+    }
+  `);
+  const sdkPackage = runnerWithCore.context.sdkPackage;
+  strictEqual(sdkPackage.clients.length, 0);
```

Comment:
can you also make sure if there's a model-only namespace with explicit usage marked that it's filtered out from the clients list, but not filtered out from `.namespaces` and that model is still exposed? Thanks!

Reply:
test updated